### PR TITLE
Add Netlify CMS

### PR DESCRIPTION
**Pull request by thisisdano**

### CODE REVIEW

File: admin/config.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+backend:
+  name: github
+  repo: uswds/uswds
```
Comment (thisisdano) [inline admin/config.yml]: This should be `uswds/uswds-site`

File: admin/config.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  branch: master
+  base_url: https://designsystem.digital.gov/
```
Comment (thisisdano) [inline admin/config.yml]: Maybe try `https://federalistapp.18f.gov`

File: admin/config.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  auth_endpoint: external/auth/github
+
```
Comment (thisisdano) [inline admin/config.yml]: try adding `preview_context: federalist/build`?

**Comment by mejiaj** [comment]
There are some pages like [about/research.md](https://github.com/uswds/uswds-site/blob/master/pages/about/research.md) that have duplicate `redirect_from` which seems to cause errors like this in netlify admin panel:

<details>
 <summary>About admin page console error</summary>

```js
{…}
​
mark: Object { buffer: "\npermalink: /about/research/\nredirect_from:\n- /about-our-work/research-strategy/\n- /about/research-strategy/\nlayout: styleguide\ntitle: Research\ncategory: About USWDS\nlead: User research is a core aspect of USWDS as it’s our main source of feedback and inspiration for future product development.\nredirect_from:\n  - /whats-new/research/\n\u0000", position: 296, line: 9, … }
​
message: "duplicated mapping key at line 10, column -39:\n    redirect_from:\n    ^"
​
name: "YAMLException"
​
reason: "duplicated mapping key"
​
stack: "r@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:33:54448\nT@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:37242\nC@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:37340\nM@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:39238\nB/<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:43711\nB@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:43962\nU@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:50752\nq@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:51404\nW@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:51554\ne.exports.safeLoad@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:51883\nfromFile@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:48:270600\nparse@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:94961\ne.exports@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:104:1181\nf/<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:96907\nf@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:97167\nfromFile@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:95621\ne.exports@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:40:89704\na<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:48:268079\ne.exports@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:40:89757\ne.exports/<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:48:268394\nentryWithFormat/<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:171508\nprocessEntries@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:167273\nlistEntries/<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:167788\npromise callback*listEntries@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:167765\nq/<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:119642\nr/</</<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:2725\nc/<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:195617\ncomponentDidUpdate@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:123:64649\nos@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:83497\npl@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:101222\nt.unstable_runWithPriority@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:84:3844\nqi@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:45071\nfl@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:97696\nZs@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:93899\nGi/<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:45362\nt.unstable_runWithPriority@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:84:3844\nqi@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:45071\nGi@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:45309\nVi@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:45242\nne@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:114302\nYt@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:22765\nEventListener.handleEvent*$t@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:22684\nKt@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:22486\nht@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:18078\nln@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:25758\nQa@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:76791\ncl@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:96877\nul@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:96766\nsl@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:96619\nZs@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:93700\nGi/<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:45362\nt.unstable_runWithPriority@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:84:3844\nqi@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:45071\nGi@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:45309\nVi@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:45242\n$s@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:90485\nenqueueSetState@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:49027\n_.prototype.setState@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:68:1297\nT/</s</l.onStateChange@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:5363\nnotify@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:2042\nw</t.notifyNestedSubs@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:1684\nT/</s</l.onStateChange@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:5381\ng@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:193549\nt.waitUntilAction/</<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:48:264710\nr/</</<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:2734\ndispatch@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:197275\n_/<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:182460\nr/</</<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:2725\ndispatch@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:197275\nt.loadConfig/<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:180174\nasync*r/</</<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:92:2725\nc/<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:28:195617\ncomponentDidMount@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:119:6157\nos@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:83398\npl@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:101222\nt.unstable_runWithPriority@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:84:3844\nqi@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:45071\nfl@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:97696\nZs@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:93899\n$s@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:90464\nLl@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:111842\nql/<@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:113087\ntl@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:94047\nql@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:113072\nt.render@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:76:116055\n_@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:68:12228\n@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:183:338408\nn@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:1:349\n@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:1:1148\n@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:1:1160\n@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:1:210\n@https://unpkg.com/netlify-cms@%5E2.0.0/dist/netlify-cms.js:1:214\n"
​
<prototype>: Object { constructor: r(e, t), toString: toString(e), stack: "" }
backend.js:510:50

```
</details>

We also need a proper testing environment for more advanced changes like redirects and setting permalinks.

**Comment by thisisdano** [comment]
We won't be doing this

**Pull request closed without merging**